### PR TITLE
[AIRFLOW-XXX] add note warning that bash>4.0 is required for docs build

### PR DESCRIPTION
**Pull request by dstandish**
If you try to run docs/build.sh script on bash included with macOS you wil get error that `mapfile` command is not present.

This is because macOS comes with bash 3.2.  You need >= 4.0 to run this script.

Adding this note can save mac users some confusion.

They need to run `brew install bash` and enable the newer version in `/etc/shells`.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: CONTRIBUTING.rst
Comment (potiuk) [inline CONTRIBUTING.rst:99]: @dstandish: Or maybe we should tell people to use `breeze` instead? (we can wait with that until I merge some simplifications and optimisations to make it quite a bit lighter). 

Breeze has stabilised a lot and it is being used by more people now so I think that might be good time to make it the "default" dev environment.

We have `./breeze --build-docs` command that runs the build in automatically downloaded/updated docker container so no matter what  OS you run it all it will yield the same results (note that it is written in the way that it will generate the output files in host (the volumes are mounted from host during build). And you do not need the doc server in fact as you can simply open the index.html locally and it will work the same (except search maybe).

Some more context: I have plans to remove all the locally run scripts from the documentation and leave breeze as the default way of doing most development tasks. I am going to remove most of the docs referring to the scripts and leave breeze docs only.

WDYT @dstandish ? We can add this comment now, but maybe going 'breeze-first' should be the next step ?

**Review by potiuk — APPROVED** [review]

File: CONTRIBUTING.rst
Comment (dstandish) [reply CONTRIBUTING.rst:99]: yeah i totally get the value of using breeze for everything ... and that's what i used initially... just when i noticed that there was a host script it seemed significantly faster than docker so that's what i started using.  i try to stay in virtualenv when i can for this reason.

but yesterday when i moved to a new machine, i forgot that i needed to upgrade bash again and stumbled around for a while (googling how to install mapfile etc).  so that was the impetus for adding.

but i defer to you on this completely -- whatever you think is best.  if you are getting rid of host build script soon then yeah probably makes makes sense not to bother.

thanks

Comment (potiuk) [reply CONTRIBUTING.rst:99]: It's perfectly OK to merge it now. I just wanted to hear your opinion on breeze's usage in this case. I will make some changes soon to make Breeze faster/more appealing and then I will remove all the reference to scripts :)

**Pull request merged**
